OpenRA.Mods.Cnc/UtilityCommands/ImportTSMapCommand.cs

```diff
@@ -556,29 +565,92 @@ static void ReadActors(Map map, IniFile file, string type, int2 fullSize)
 
 		static void ReadLighting(Map map, IniFile file)
 		{
-			var lightingTypes = new[] { "Red", "Green", "Blue", "Ambient" };
+			var lightingTypes = new Dictionary<string, string>()
+			{
+				{ "Red", "RedTint" },
+				{ "Green", "GreenTint" },
+				{ "Blue", "BlueTint" },
+				{ "Ambient", "Intensity" },
+				{ "Level", "HeightStep" },
+				{ "Ground", null }
+			};
+
 			var lightingSection = file.GetSection("Lighting");
+			var parsed = new Dictionary<string, float>();
 			var lightingNodes = new List<MiniYamlNode>();
 
 			foreach (var kv in lightingSection)
 			{
-				if (lightingTypes.Contains(kv.Key))
-				{
-					var val = FieldLoader.GetValue<float>(kv.Key, kv.Value);
-					if (val != 1.0f)
-						lightingNodes.Add(new MiniYamlNode(kv.Key, FieldSaver.FormatValue(val)));
-				}
+				if (lightingTypes.ContainsKey(kv.Key))
+					parsed[kv.Key] = FieldLoader.GetValue<float>(kv.Key, kv.Value);
 				else
 					Console.WriteLine("Ignoring unknown lighting type: `{0}`".F(kv.Key));
 			}
 
+			// Merge Ground into Ambient
+			float ground = 0;
+			if (parsed.TryGetValue("Ground", out ground))
+			{
+				if (!parsed.ContainsKey("Ambient"))
+					parsed["Ambient"] = 1f;
+				parsed["Ambient"] -= ground;
+			}
+
+			foreach (var node in lightingTypes)
+			{
+				float val;
+				if (node.Value != null && parsed.TryGetValue(node.Key, out val) && ((node.Key == "Level" && val != 0) || (node.Key != "Level" && val != 1.0f)))
+					lightingNodes.Add(new MiniYamlNode(node.Value, FieldSaver.FormatValue(val)));
+			}
+
 			if (lightingNodes.Any())
 			{
 				map.RuleDefinitions.Nodes.Add(new MiniYamlNode("World", new MiniYaml("", new List<MiniYamlNode>()
 				{
-					new MiniYamlNode("GlobalLightingPaletteEffect", new MiniYaml("", lightingNodes))
+					new MiniYamlNode("^BaseWorld", new MiniYaml("", lightingNodes))
```

Comment:
Nts: Test this, since it looks like this should actually be `TerrainLighting` here and `^BaseWorld` above.